src/transformers/onnx/config.py

```diff
@@ -197,40 +211,60 @@ def use_external_data_format(num_parameters: int) -> bool:
 
     def generate_dummy_inputs(
         self,
-        tokenizer: PreTrainedTokenizer,
+        preprocessor: Union[PreTrainedTokenizer, FeatureExtractionMixin],
         batch_size: int = -1,
         seq_length: int = -1,
         is_pair: bool = False,
         framework: Optional[TensorType] = None,
+        tokenizer: PreTrainedTokenizer = None,
```

Comment:
This is an example of a function whose signature needed to be changed to:

* Replace the positional `tokenizer` argument with a more generic `preprocessor` one
* Retain the `tokenizer` as a keyword argument - I'll add a warning that this usage is deprecated

Reply:
If I understand correctly the tokenizer keyword argument is to maintain backward compatibility?

Reply:
Yes, that's correct. I'd like to deprecate this in v5 of `transformers`, so will add a warning to alert the user